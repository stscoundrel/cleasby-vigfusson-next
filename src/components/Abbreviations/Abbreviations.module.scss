.column {
  margin-bottom: 30px;
}

@media only screen and (min-width: 768px) {
  .abbreviations {
    display: flex;
    margin-top: 50px;
  }

  .column {
    flex-basis: 50%;
    width: 50%;
  }
}