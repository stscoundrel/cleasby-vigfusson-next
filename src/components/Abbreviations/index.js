import AbbreviationList from 'components/AbbreviationList'
import styles from './Abbreviations.module.scss'

export default function Abbreviations({ abbreviations }) {
  const { common: commonAbbreviations, works: workAbbreviations } = abbreviations

  return (
    <div className={styles.abbreviations}>
      {commonAbbreviations.length > 0
        && <div className={styles.column}>
          <h4>Abbreviations used:</h4>
          <AbbreviationList abbreviations={commonAbbreviations} />
        </div>
      }
      {workAbbreviations.length > 0
        && <div className={styles.column}>
          <h4>Works & Authors cited:</h4>
          <AbbreviationList abbreviations={workAbbreviations} />
        </div>
      }
    </div>
  )
}
