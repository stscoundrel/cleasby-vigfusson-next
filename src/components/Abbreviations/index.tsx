import AbbreviationList from 'components/AbbreviationList'
import { CombinedAbbreviations } from 'lib/services/abbreviations'
import styles from './Abbreviations.module.scss'

interface AbbreviationProps{
  abbreviations: CombinedAbbreviations
}

export default function Abbreviations({ abbreviations }: AbbreviationProps) {
  const {
    common: commonAbbreviations,
    works: workAbbreviations,
  } = abbreviations

  return (
    <div className={styles.abbreviations}>
      {commonAbbreviations.length > 0
        && <div className={styles.column}>
          <h4>Abbreviations used:</h4>
          <AbbreviationList abbreviations={commonAbbreviations} />
        </div>
      }
      {workAbbreviations.length > 0
        && <div className={styles.column}>
          <h4>Works & Authors cited:</h4>
          <AbbreviationList abbreviations={workAbbreviations} />
        </div>
      }
    </div>
  )
}
