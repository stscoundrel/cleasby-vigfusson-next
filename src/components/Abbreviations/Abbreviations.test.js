import React from 'react'
import renderer from 'react-test-renderer'
import Abbreviations from './index'

describe('Abbreviations component', () => {
  const abbreviations = {
    common: [
      {
        abbreviation: 'f.',
        explanation: 'feminine.',
      },
      {
        abbreviation: 'n.',
        explanation: 'neuter.',
      },
    ],
    works: [
      {
        abbreviation: 'Björn',
        explanation: 'Biörn Halldórsson.',
      },
    ],
  }

  test('Matches the snapshot', () => {
    const tree = renderer.create(
      <Abbreviations abbreviations={abbreviations} />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
