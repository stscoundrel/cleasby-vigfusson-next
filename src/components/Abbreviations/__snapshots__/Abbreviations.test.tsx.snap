// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Abbreviations component Matches the snapshot 1`] = `
<div
  className="abbreviations"
>
  <div
    className="column"
  >
    <h4>
      Abbreviations used:
    </h4>
    <dl
      className="wrap"
    >
      <dt
        className="abbreviation"
      >
        <strong>
          f.
        </strong>
      </dt>
      <dd
        className="explanation"
      >
        feminine.
      </dd>
    </dl>
    <dl
      className="wrap"
    >
      <dt
        className="abbreviation"
      >
        <strong>
          n.
        </strong>
      </dt>
      <dd
        className="explanation"
      >
        neuter.
      </dd>
    </dl>
  </div>
  <div
    className="column"
  >
    <h4>
      Works & Authors cited:
    </h4>
    <dl
      className="wrap"
    >
      <dt
        className="abbreviation"
      >
        <strong>
          Björn
        </strong>
      </dt>
      <dd
        className="explanation"
      >
        Biörn Halldórsson.
      </dd>
    </dl>
  </div>
</div>
`;
